src/Ng/RegistrationStorageFactories.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace Ng
+{
+    public class RegistrationStorageFactories
```

Comment:
XML comments in this file?

I know you have them in `Catalog2Registration`'s `Init`, but I think they belong here now that you've shared the storage factory creation logic.

```
/// <summary>
/// Contains the storage factories used to create registration blobs.
/// </summary>
public class RegistrationStorageFactories
```

```
/// <summary>
/// Supports legacy clients which cannot use the current registration blobs.
/// Maintains the cursor of the job.
/// Contains an aggregate storage factory that writes two sets of registration blobs.
/// 1 - not gzipped without SemVer 2.0.0 packages
/// 2 - gzipped without SemVer 2.0.0 packages
/// </summary>
public StorageFactory LegacyStorageFactory { get; }

/// <summary>
/// Writes registration blobs that are gzipped and contain SemVer 2.0.0 packages
/// </summary>
public StorageFactory SemVer2StorageFactory { get; }
```